webui/react/src/components/UPlot/SyncableBounds.tsx

```diff
@@ -0,0 +1,150 @@
+// @ts-nocheck
+import React, {
+  createContext,
+  MutableRefObject,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import uPlot from 'uplot';
+
+import { distance } from 'utils/chart';
+
+const timestampInMinutes = t => new Date(t).toTimeString().slice(3, 9);
+
+interface SyncableBounds {
+  boundsOptions: Partial<uPlot.Options>;
+  isZoomed: (zoomed: boolean) => void;
+  zoomed: boolean;
+}
+
+const SyncContext = createContext();
+
+const getChartMin = (chart: UPlot) => chart.scales.x.min;
+const getChartMax = (chart: UPlot) => chart.scales.x.max;
+
+export const SyncProvider = ({ children }) => {
+  const syncRef = useRef(uPlot.sync('x'));
+  const [ syncedMin, setSyncedMin ] = useState();
+  const [ syncedMax, setSyncedMax ] = useState();
+  const [ queueSignal, setQueueSignal ] = useState(0);
+  const [ zoomed, setZoomed ] = useState(false);
+  const fireQueue = useCallback(
+    () => setQueueSignal(prevSignal => (prevSignal + 1) % 1000)
+    , [ setQueueSignal ],
+  );
+
+  const minMaxRef = useRef({});
+
+  const updateQueue = useRef([]);
```

Comment:
Why use a ref and a queue vs setState? I guess I was concerned about things happening in the wrong order, or something setting a new max based on an outdated existing max, etc.

I guess a setState(prevMax => max(prevMax, chartMax)) would be fine though, and we wouldn't need the queue?